uid/src/main/java/com/tomtom/kotlin/uid/Uid.kt

```diff
@@ -0,0 +1,225 @@
+/*
+ * Copyright (C) 2020-2020, TomTom (http://tomtom.com).
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.tomtom.kotlin.uid
+
+import java.util.*
+
+/**
+ * Generic immutable Unique ID class. Really just an abstraction of UUIDs. Used to unique
+ * identify things. No 2 Uid objects shall be equal unless they are the same instance of the
+ * Uid class or their underlying UUIDs are equal.
+ *
+ * The class is templated to allow creating typesafe Uid's, like Uid<Message> or Uid<Person>.
```

Comment:
template is C++. It should be: `This class is a generic type to...`

Reply:
Oops, sorry. Changed.